File: 06. Matrix/javascript/readMatrix.js
// Problem Description:

// Access and print matrix elements in same matrix representation.

function readMatrix(matrix) {

    let row = matrix.length;
    let col = matrix[0].length;

    for(let i=0; i<row; i++) {
        let rowString = "";
        for(let j=0; j<col; j++) {
            rowString += matrix[i][j] + " ";
        }
        console.log(rowString);
    }
}

readMatrix([[1,2,3], [4,5,6], [7,8,9]])

/* 
    Time Complexity : O(n*m)
    Space Complexity : O(1)
*/

File: 06. Matrix/python/readMatrix.py
# Problem Description:

# Access and print matrix elements in same matrix representation.

def readMatrix(matrix):

    row = len(matrix)
    col = len(matrix[0])

    for i in range(row):
        for j in range(col):
            print(matrix[i][j], end=" ")
        print()

print("Enter your matrix row by row (press Enter on a blank line to finish):")

matrix = []
while True:
    line = input()
    if not line.strip():  # Stops reading if the user enters a blank line
        break
    
    # Split the line by spaces, convert each string to an integer, and create a list
    row = [int(x) for x in line.split()]
    matrix.append(row)

readMatrix(matrix);

# Time Complexity : O(n*m)
# Space Complexity : O(1)